packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -298,14 +306,42 @@ export class PendingStateManager implements IDisposable {
 	 * that the batch information is correct.
 	 * @param batch - The batch that is being processed.
 	 * @param batchStartCsn - The clientSequenceNumber of the start of this message's batch
+	 * @param sequenceNumber - The sequence number of the batch message.
 	 */
 	public processPendingLocalBatch(
 		batch: InboundSequencedContainerRuntimeMessage[],
 		batchStartCsn: number,
+		sequenceNumber?: number,
 	): {
 		message: InboundSequencedContainerRuntimeMessage;
 		localOpMetadata: unknown;
 	}[] {
+		if (batch.length === 0) {
+			const pendingMessage = this.pendingMessages.peekFront();
+			assert(pendingMessage !== undefined, "No pending message found for this remote message");
+			assert(pendingMessage.opMetadata?.emptyBatch === true, "Expected empty batch");
+			assert(sequenceNumber !== undefined, "Expected sequence number for empty batch");
+
+			if (pendingMessage.batchIdContext.batchStartCsn !== batchStartCsn) {
```

Comment:
Can you just call `maybeProcessBatchBegin` here to stay DRY?  Or, in my next PR I'm moving this check to this function anyway, you can probably just do that in this PR (do it at the very top)